qemu/tests/multi_nics_verify.py

```diff
@@ -35,39 +42,68 @@ def check_nics_num(expect_c, session):
-    error_context.context(txt, logging.info)
-    ifname_list = utils_net.get_linux_ifname(session)
-    ifcfg_path = "/etc/sysconfig/network-scripts/ifcfg-%s"
-    for ifname in ifname_list:
-        eth_config_path = ifcfg_path % ifname
-        eth_config = "DEVICE=%s\\nBOOTPROTO=dhcp\\nONBOOT=yes" % ifname
-        cmd = "echo -e '%s' > %s" % (eth_config, eth_config_path)
-        s, o = session.cmd_status_output(cmd)
-        if s != 0:
-            err_msg = "Failed to create ether config file: %s\nReason is: %s"
-            test.error(err_msg % (eth_config_path, o))
+    os_type = params.get("os_type", "linux")
+    if os_type == "linux":
+        # Redirect ifconfig output from guest to log file
+        log_file = os.path.join(test.debugdir, "ifconfig")
+        ifconfig_output = session.cmd("ifconfig")
+        log_file_object = open(log_file, "w")
+        log_file_object.write(ifconfig_output)
+
+        # Pre-judgement for the ethernet interface
+        logging.debug(check_nics_num(nics_num, session)[1])
+        txt = "Create configure file for every NIC interface in guest."
+        error_context.context(txt, logging.info)
+        ifname_list = utils_net.get_linux_ifname(session)
+        ifcfg_path = "/etc/sysconfig/network-scripts/ifcfg-%s"
+        for ifname in ifname_list:
+            eth_config_path = ifcfg_path % ifname
+            eth_config = "DEVICE=%s\\nBOOTPROTO=dhcp\\nONBOOT=yes" % ifname
+            cmd = "echo -e '%s' > %s" % (eth_config, eth_config_path)
+            s, o = session.cmd_status_output(cmd)
+            if s != 0:
+                err_msg = "Failed to create ether config file: %s\nReason is: %s"
+                test.error(err_msg % (eth_config_path, o))
+        log_file_object.close()
+
+        # Reboot and check the configurations.
+        new_session = vm.reboot(session)
+        s, msg = check_nics_num(nics_num, new_session)
+        if not s:
+            test.fail(msg)
+        session.close()
 
-    # Reboot and check the configurations.
-    new_session = vm.reboot(session)
-    s, msg = check_nics_num(nics_num, new_session)
-    if not s:
-        test.fail(msg)
+        # NICs matched.
+        logging.info(msg)
 
-    # NICs matched.
-    logging.info(msg)
+    # Check all the interfaces in guest get ips
+    nic_interface = []
+    session_srl = vm.wait_for_serial_login(timeout=int(params.get("login_timeout", 360)))
+    for index, nic in enumerate(vm.virtnet):
+        logging.info("index %s nic" % index)
+        guest_ip = utils_net.get_guest_ip_addr(session_srl, nic.mac, os_type,
+                                               ip_version="ipv4")
+        if not guest_ip:
+            err_log = "vm get interface %s's ip failed." % index
+            test.fail(err_log)
+        nic_interface.append(guest_ip)
+    nic_interface_len = len(nic_interface)
+    session_srl.close()
+    if nics_num != nic_interface_len:
```

Comment:
You have already raise `test.fail` in line102 when nic doesn't get an ip address.

Reply:
@sitoliu @pezhang 
Updated. Please help review again. Thanks.